### PR TITLE
feat: Add `and`, `nor` operators in QueryBuilder

### PR DESCRIPTION
**Pull request by mbfakourii**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description
add ```and``` and ```nor```
add ```_constructorInitializer``` function for handle ```and``` , ```nor``` and ```or``` in QueryBuilder
add three tests for and, nor, or
Related issue: #796

### Approach
The value of ```and``` and ```nor```  was not present in this package

### TODOs before merging

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] A changelog entry

### CODE REVIEW

**Review by mtrezza — CHANGES_REQUESTED** [review]
Could you add tests? There is also a lint issue.

**Comment by mbfakourii** [comment]
> Could you add tests? There is also a lint issue.

I added three tests for and, nor, or

**Comment by mtrezza** [comment]
I don't think you need to add anything to README, what would you add?

**Comment by mbfakourii** [comment]
> I don't think you need to add anything to README, what would you add?

documentation for how to use and and nor

**Comment by mtrezza** [comment]
Oh, if that's not there yet, it would be great if you could add it!

File: packages/dart/README.md [hunk 1]
```diff
@@ -312,6 +312,7 @@ QueryBuilder<ParseObject> mainQuery = QueryBuilder.or(
 
 var apiResponse = await mainQuery.query();
 ```
+and also if you want to find objects that match several queries you can use __QueryBuilder.and__  and If you want to find objects that do not match all queries, you can use __QueryBuilder.nor__
```
Comment (mtrezza) [inline packages/dart/README.md:315]:
```suggestion

To find objects that match several queries use __QueryBuilder.and__ . To find objects that do not match any given query use __QueryBuilder.nor__.
```

File: packages/flutter/README.md [hunk 2]
```diff
@@ -350,6 +350,7 @@ QueryBuilder<ParseObject> mainQuery = QueryBuilder.or(
 
 var apiResponse = await mainQuery.query();
 ```
+and also if you want to find objects that match several queries you can use __QueryBuilder.and__  and If you want to find objects that do not match all queries, you can use __QueryBuilder.nor__
```
Comment (mtrezza) [inline packages/flutter/README.md:353]:
```suggestion

To find objects that match several queries use __QueryBuilder.and__. To find objects that do not match any given query use __QueryBuilder.nor__.
```

**Comment by mbfakourii** [comment]
I use ```and``` in a project that is used by several people every day. It seems that there is no problem

**Comment by mtrezza** [comment]
Rerunning failed tests

**Comment by mbfakourii** [comment]
> Rerunning failed tests

What is the problem, like all the tests are passed

**Comment by mtrezza** [comment]
Now they did, earlier they did not, probably some flaky tests.

**Comment by mtrezza** [comment]
Could you again bump the versions and add the changelog entry?

File: packages/dart/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+## [3.1.6](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.5...dart-3.1.6) (2022-12-21)
+
+### Bug Fixes
+
+* Add and, nor operators in QueryBuilder ([#795](https://github.com/parse-community/Parse-SDK-Flutter/issues/795))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
* Add `and`, `nor` operators in QueryBuilder ([#795](https://github.com/parse-community/Parse-SDK-Flutter/issues/795))
```

**Review by mtrezza — APPROVED** [review]

**Pull request merged**